src/server/templates/target.js

```diff
@@ -69,7 +69,7 @@ export default ({
 			<div id='target'>${markup}</div>
 			<script src='/js/bundle.js'></script>
 			${page ? pageScript : ''}
-			${props && script &&
+			${props !== null && script !== null ? /** checking if the variables are undefined or null */
```

Comment:
Good initiative to document the code !
I think our automated tools will complain and require the comment line to be on a separate line above the code.

Also, your comment says you're checking for  "undefined or null", but the code only checks for null, and not undefined, as they are different in javascript.
You'll need to check for both null and undefined: `props !== null && typeof props !== "undefined" && …`

This code is starting to be a bit verbose at this point, so I would move the logic before the `return` statement on line 51.
Something like (in pseudo-code):
```
let scriptTags = null;
if(props !== null && typeof props !== "undefined" && MY OTHER CONDITIONS){
  scriptTags = `<script .......`
}
return (
    .....
    {scriptTags}
  </body>
  .....
```
We instantiate `scriptTags` as `null`, so if the logic in the if block is not executed, it will stay as `null`.
React will interpret that correctly and simply not display anything !

Reply:
Hi @MonkeyDo , 

Thank you so much for your help!! :DD